### PR TITLE
VACMS 16019 KISS VAMC System/facilities list

### PR DESCRIPTION
**Pull request by eselkin**
## Summary

- Gets CSV data from CMS using curl process. Transforms CSV to JSON with structure most usable for vets-website
- KISS JSON static file created in build
- Sitewide Facilities

## Related issue(s)

- https://github.com/department-of-veterans-affairs/va.gov-cms/issues/16019

## Testing done

- Added test (postProcess step after download of CSV from CMS). Test passes.
- Added tests (existence of json post build with e2e). Tests pass.

## Screenshots

N/A

## What areas of the site does it impact?

Adds JSON to static files built.

## Acceptance criteria

- [x] CSV is pulled from prod CMS
- [x]  vamc-system.json is created in cache static data files
- [x]  e2e testing is added that vamc-system.json is created

### Quality Assurance & Testing

- [x] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [x] Linting warnings have been addressed
- [x] Documentation has been updated (No new documentation)
- [x] Screenshot of the developed feature is added (N/A)
- [x] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) (N/A)

### Error Handling

- [x] Browser console contains no warnings or errors.
- [x] Events are being sent to the appropriate logging solution
- [x] Feature/bug has a monitor built into Datadog or Grafana (N/A)

### Authentication

- [x] No routes impacted

## Requested Feedback

Test if `.cache/localhost/drupal/static-data-files/vamc-system.json` is created.
You can test in a built environment with `http://localhost:3002/data/cms/vamc-system.json` or in a tugboat at that path (https://web-ihqrv6rby5lxlkn8zqdxf9u7lk1aplcp.demo.cms.va.gov/data/cms/vamc-system.json).

### CODE REVIEW

File: src/site/stages/build/drupal/tests/system/process.unit.spec.js [hunk 1]
```diff
@@ -0,0 +1,43 @@
+/* eslint-disable camelcase */
```
[the remark this answers is not in the document]
Comment (eselkin) [reply src/site/stages/build/drupal/tests/system/process.unit.spec.js:1]: data test object has snake case "vha_" prefix.

File: src/site/stages/build/drupal/tests/system/process.unit.spec.js [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+import fs from 'fs-extra';
+import path from 'path';
+import { expect } from 'chai';
+import { postProcessVamcSystems } from '../../static-data-files/vamcSystem/postProcessVamcSystem';
+
+describe('postProcessVamcSystems', () => {
+  // Not a cypress test, but a unit test
+  // eslint-disable-next-line @department-of-veterans-affairs/axe-check-required
```
[the remark this answers is not in the document]
Comment (eselkin) [reply src/site/stages/build/drupal/tests/system/process.unit.spec.js:9]: Not a cypress test

File: src/site/stages/build/drupal/tests/police/postProcess.unit.spec.js [hunk 3]
```diff
@@ -0,0 +1,76 @@
+/* eslint-disable camelcase */
+/* eslint-disable @department-of-veterans-affairs/axe-check-required */
```
[the remark this answers is not in the document]
Comment (eselkin) [reply src/site/stages/build/drupal/tests/police/postProcess.unit.spec.js:2]: Data requires snake case prefixes "vha_"

File: src/site/stages/build/drupal/tests/police/postProcess.unit.spec.js [hunk 4]
```diff
@@ -0,0 +1,76 @@
+/* eslint-disable camelcase */
```
[the remark this answers is not in the document]
Comment (eselkin) [reply src/site/stages/build/drupal/tests/police/postProcess.unit.spec.js:1]: Data requires snake case "vha_" prefixes

File: src/site/stages/build/drupal/static-data-files/vamcSystem/postProcessVamcSystem.js [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+    if (!processedJSON.data.systems[jsonFacility['VAMC system']]) {
+      processedJSON.data.systems[jsonFacility['VAMC system']] = {
+        [jsonFacility['Facility Locator API ID']]:
+          jsonFacility['VAMC facility name'],
+      };
+    } else {
+      processedJSON.data.systems[jsonFacility['VAMC system']][
+        jsonFacility['Facility Locator API ID']
+      ] = jsonFacility['VAMC facility name'];
+    }
```
Comment (maxx1128) [inline src/site/stages/build/drupal/static-data-files/vamcSystem/postProcessVamcSystem.js:21]: **Question:** I'm not entirely clear on what this conditional checks and how it creates the outputted data.

**Review by maxx1128 — APPROVED** [review]
The code looks good and the tests are passing. I have one question but the answer is not a blocker.

**Review by rsmithadhoc — APPROVED** [review]

**Pull request merged**